src/content-tags-drawer/ContentTagsDropDownSelector.jsx

```diff
@@ -0,0 +1,99 @@
+import React, { useState } from 'react';
+import {
+  SelectableBox,
+  Icon,
+  Spinner,
+} from '@edx/paragon';
+import { useIntl } from '@edx/frontend-platform/i18n';
+import { ArrowDropDown, ArrowDropUp } from '@edx/paragon/icons';
+import PropTypes from 'prop-types';
+import messages from './messages';
+import './ContentTagsDropDownSelector.scss';
+
+import { useTaxonomyTagsDataResponse, useIsTaxonomyTagsDataLoaded } from './data/apiHooks';
+
+const ContentTagsDropDownSelector = ({
+  taxonomyId, level, subTagsUrl,
+}) => {
+  const intl = useIntl();
+  // This object represents the states of the dropdowns on this level
+  // The keys represent the index of the dropdown with
+  // the value true (open) false (closed)
+  const [dropdownStates, setDropdownStates] = useState({});
+
+  const isOpen = (i) => dropdownStates[i];
+
+  const clickAndEnterHandler = (i) => {
+    // This flips the state of the dropdown at index false (closed) -> true (open)
+    // and vice versa. Initially they are undefined which is falsy.
+    setDropdownStates({ ...dropdownStates, [i]: !dropdownStates[i] });
+  };
+
+  const useTaxonomyTagsData = (id, fullPathProvided) => {
+    const taxonomyTagsData = useTaxonomyTagsDataResponse(id, fullPathProvided);
+    const isTaxonomyTagsLoaded = useIsTaxonomyTagsDataLoaded(id, fullPathProvided);
+    return { taxonomyTagsData, isTaxonomyTagsLoaded };
+  };
+
+  const { taxonomyTagsData, isTaxonomyTagsLoaded } = useTaxonomyTagsData(taxonomyId, subTagsUrl);
```

Comment:
Won't it be shorter to remove the function call here?

```suggestion
  const taxonomyTagsData = useTaxonomyTagsDataResponse(taxonomyId, subTagsUrl);
  const isTaxonomyTagsLoaded = useIsTaxonomyTagsDataLoaded(taxonomyId, subTagsUrl);
```

Reply:
Yup! Updated it here: https://github.com/openedx/frontend-app-course-authoring/pull/654/commits/43b2de57da7bfcd8b3442d7b41cb961c76ab79ff